File: kotlin/0448-e-find-all-numbers-disappeared-in-an-array.kt
// find-all-numbers-disappeared-in-an-array
class Solution {
    fun findDisappearedNumbers(nums: IntArray): List<Int> {
        val existNums = IntArray(nums.size+1) { 0 }

        for (n in nums) {
            existNums[n] = 1
        }
        
        val result = mutableListOf<Int>()

        for (i in 1..nums.size) {
            if (existNums[i] == 0) {
                result.add(i)
            }
        }

        return result
    }
}

File: javascript/0448-e-find-all-numbers-disappeared-in-an-array.js
// https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    const result = [];
    const numsSet = new Set(nums);

    for(let i = 1; i < nums.length+1; i = i+1) {
        if (!numsSet.has(i)) {
            result.push(i);
        }
    }

    return result;
};